packages/material-ui/src/TablePagination/TablePagination.js

```diff
@@ -1,75 +1,135 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import { chainPropTypes, integerPropType } from '@material-ui/utils';
 import clsx from 'clsx';
-import withStyles from '../styles/withStyles';
+import { chainPropTypes, integerPropType } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses, isHostComponent } from '@material-ui/unstyled';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
 import InputBase from '../InputBase';
 import MenuItem from '../MenuItem';
 import Select from '../Select';
 import TableCell from '../TableCell';
 import Toolbar from '../Toolbar';
-import Typography from '../Typography';
 import TablePaginationActions from './TablePaginationActions';
 import useId from '../utils/useId';
+import tablePaginationClasses, { getTablePaginationUtilityClass } from './tablePaginationClasses';
+
+const TablePaginationRoot = experimentalStyled(
+  TableCell,
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Root',
+    overridesResolver: (props, styles) => styles.root,
+  },
+)(({ theme }) => ({
+  overflow: 'auto',
+  color: theme.palette.text.primary,
+  fontSize: theme.typography.pxToRem(14),
+  // Increase the specificity to override TableCell.
+  '&:last-child': {
+    padding: 0,
+  },
+}));
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    color: theme.palette.text.primary,
-    fontSize: theme.typography.pxToRem(14),
-    overflow: 'auto',
-    // Increase the specificity to override TableCell.
-    '&:last-child': {
-      padding: 0,
-    },
+const TablePaginationToolbar = experimentalStyled(
+  Toolbar,
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Toolbar',
+    overridesResolver: (props, styles) => styles.toolbar,
   },
-  /* Styles applied to the Toolbar component. */
-  toolbar: {
+)(({ theme }) => ({
+  minHeight: 52,
+  paddingRight: 2,
+  [`${theme.breakpoints.up('xs')} and (orientation: landscape)`]: {
```

Comment:
Where are these styles coming from?

Reply:
same as above, to override Toolbar styles

Reply:
I think let's keep it because the PR is only about migration.